Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -473,8 +355,8 @@ void Shutdown()
 void SetDisc(std::unique_ptr<DiscIO::VolumeDisc> disc,
              std::optional<std::vector<std::string>> auto_disc_change_paths = {})
 {
-  if (disc)
-    s_current_partition = disc->GetGamePartition();
+  bool had_disc = IsDiscInside();
+  bool has_disc = (bool)disc;
```

Comment:
Thoughts on using `bool has_disc = static_cast<bool>(disc)` or the infamous cast-to-bool operator `bool has_disc = !!disc` instead of the C-style cast?

Reply:
I find the `!!` thing somewhat silly looking, so I'd probably use static_cast instead.  I probably could also do just an implicit conversion (`bool has_disc = disc`) but that seems like it'd also be confusing.